Sources/SpeziOnboarding/OnboardingFlow/OnboardingNavigationPath.swift

```diff
@@ -0,0 +1,240 @@
+/// ```
+@MainActor
+public class OnboardingNavigationPath: ObservableObject {
+    /// Internal SwiftUI `NavigationPath` that serves as the source of truth for the navigation state.
+    /// Holds elements of type `OnboardingStepIdentifier` which identify the individual onboarding steps.
+    @Published var path = NavigationPath()
+    /// Boolean binding that is injected via the ``OnboardingStack``.
+    /// Indicates if the onboarding flow is completed, meaning the last view declared within the ``OnboardingStack`` is completed.
+    private var complete: Binding<Bool>?
+    
+    /// Stores all onboarding views as declared within the ``OnboardingStack``.
+    private var onboardingSteps: [OnboardingStepIdentifier: any View] = [:]
+    /// Stores all custom onboarding views that are appended to the ``OnboardingNavigationPath`` via the ``append(customView:)`` or ``append(customViewInit:)`` instance methods
+    private var customOnboardingSteps: [OnboardingStepIdentifier: any View] = [:]
+    /// Stores all `OnboardingStepIdentifier`s in-order as declared by the onboarding views within the ``OnboardingStack``.
+    private var onboardingStepsOrder: [OnboardingStepIdentifier] = []
+    
+    /// The first onboarding view of the `OnboardingNavigationPath.onboardingSteps`. Serves as a starting point for the SwiftUI `NavigationStack`.
+    ///
+    /// In case there isn't a single onboarding view stored within `OnboardingNavigationPath.onboardingSteps` (meaning the ``NavigationStack`` contains no views after its evaluation), the property serves an `EmptyView` which is then dismissed immediatly as the `OnboardingNavigationPath.complete` property is automatically set to true.
+    var firstOnboardingView: AnyView {
+        guard let firstOnboardingStepIdentifier = onboardingStepsOrder.first,
+              let view = onboardingSteps[firstOnboardingStepIdentifier] else {
+            return .init(EmptyView())
+        }
+        
+        return .init(view)
+    }
+    
+    
+    /// Identifier of the current onboarding step that is shown to the user via its associated view
+    /// Inspects the `OnboardingNavigationPath.path` to determine the current on-top navigation element of the internal SwiftUI `NavigationPath`.
+    /// Utilizes the extenstion of the `NavigationPath` declared within the ``SpeziOnboarding`` package for this functionality.
+    ///
+    /// In case there isn't a suitable element within the `OnboardingNavigationPath.path`, return the `OnboardingStepIdentifier` of the first onboarding view.
+    private var currentOnboardingStep: OnboardingStepIdentifier? {
+        var copyPath = path
+        while !copyPath.isEmpty {
+            guard let lastElement = copyPath.lastElement else {
+                return nil
+            }
+            
+            if !lastElement.custom {
+                return lastElement
+            }
+            copyPath.removeLast()
+        }
+        
+        return onboardingStepsOrder.first
```

Comment:
I suspect you could remove the copy implementation using a [`last(where: ...)`](https://developer.apple.com/documentation/swift/array/last(where:)) statement?

Reply:
Sadly, there's no good way around this copying option as we are only able to access the last element of the `NavigationPath`, meaning we need to remove the on-top element in order to get to the next one. However, I moved the actual logic into `NavigationPath+Codable.swift` in order to hide the complexity from the user.

`NavigationPath+Codable.swift` now offers `last` as well as `last(where: ...)` properties/functions in the typical Swift style 🚀

Reply:
Nice, thank you!